examples/soft-kmeans/README.md

```diff
@@ -0,0 +1,108 @@
+This is highlighted by the fact that the top-left mode in the posterior shows a higher probability than the lower-right mode.
+
+
+
+## Single MCMC chain sampling
+We created a simple single MCMC chain for sampling this posterior distribution, defined in the module `single_chain.py`.
+
+### Sampling
+With an initial state for the MCMC chain set at (1, 0), the chain gets stuck on the lower probability mode.
+That means that the gaussian mixture component with the higher weight (component 2) fits the cluster with the lower number of points, and vice versa.
+
+<img alt="sampling single chain with initial state = (1, 0)" src="./images/sampling_sc_initstate-1-0.png" height="400"/>
+
+
+This results in the samples of the means being slightly offset from their expected positions (0 and 1.5), as to compensate for the mismatch.
+
+<img alt="sampled means single chain" src="./images/samples_means_sc_initstate-1-0.png" height="400"/>
+
+### Cluster assignment
+Soft k-means clustering treats the cluster assignments as probability distributions over the clusters.
+Each data point is associated with a log unnormalized probability distribution for each cluster, which is obtained from transformation of the samples of the means (called the "soft Z" transformed samples).
+These soft Z distributions, after softmax normalization, provide the probability of each point to belong to each cluster.
+The distributions of the soft Z samples for 3 individual data points are shown below (one point located on the left, one in the middle, and one on the right).
+
+<img alt="Soft_z analysis" src="./images/soft-z_samples_sc2.png"/>
+
+The leftward point at `x = -0.687`, shows a distribution of higher soft Z values for the cluster 2, that is clearly distinct for the distribution of soft Z values for cluster 1.
+Therefore, it can be unequivocally assigned to the cluster 2.
+Similarly, the rightward point at `x = 2.369` can be unequivocally assigned to cluster 1.
+However, the middle point at `x = 1.4` displays two distributions of soft Z values that are superimposed.
+This highlights the fact that this point is in an ambiguous area, and cannot be assigned confidently to any of the two cluster.
+Note that this uncertainty is not quantified in the traditional "hard" k-means algorithm, whereas it is available here, through these soft Z samples.
+
+This being said, the soft Z distributions can be summarized to their median value, and consequently, the soft k-means clustering can be converted to a "hard" k-means by assigning each point to its cluster of higher median probability.
+
+<img alt="cluster assignment single chain" src="./images/cluster_assignment_sc_initstate-1-0.png" height="400"/>
+
+In this case, because the lower probability mode of the posterior is sampled, the cluster assignment is rather imperfect.
+The points at the frontier of the 2 clusters (framed in dotted line) were assigned to the left cluster, which is the less probable one.
+
+
+
+## Chainsail sampling
+
```

Comment:
Let's summarize here in once sentence what went wrong above and why we expect Chainsail to do better.